protected/humhub/modules/comment/widgets/views/form.php

```diff
@@ -7,18 +7,25 @@
 use humhub\modules\file\widgets\UploadButton;
 use humhub\modules\file\widgets\FilePreview;
 
-/* @var $modelName string */
-/* @var $modelId integer */
+/* @var $this \humhub\modules\ui\view\components\View */
+/* @var $objectModel string */
+/* @var $objectId integer */
+/* @var $id string unique object id */
 
+/** @var \humhub\modules\content\Module $contentModule */
+$contentModule = Yii::$app->getModule('content');
 $submitUrl = Url::to(['/comment/comment/post']);
 
+// Hide the comment form for sub comments until the button is clicked
+$isHidden = ($objectModel === \humhub\modules\comment\models\Comment::class);
```

Comment:
Should be else it hides on click.
```php
$isHidden = ($objectModel !== Comment::class);
```

Reply:
Example of it working with above modification.
![Screenshot_4](https://user-images.githubusercontent.com/35392110/87365724-f5c50800-c544-11ea-957c-c88e0f134292.png)

Reply:
This completely fixed the issue where comment input disappears.

```php
$isHidden = ($objectId === Comment::class);
```

But as is if you use the `$objectModel` it will force you to only comment and the input form is completely hidden because of `===` and only shown with `!==`.
If `$objectId` is used with `===` then the issue disappears completely and you can use both inputs.

Reply:
Did you empty the cache (Browser & Assets) and rebuild the assets if necessary? Normally only the form of nested comments should be hidden.

Reply:
Yes did this and the problem stayed the same till modified.

Reply:
The current expected behavior is.

After initial stream load:
![image](https://user-images.githubusercontent.com/4736168/87522718-3f693c00-c686-11ea-8256-8e474e348c8c.png)

After click on "Comment":
![image](https://user-images.githubusercontent.com/4736168/87522792-56a82980-c686-11ea-872b-20fcb2457888.png)

Reply:
Yes, only issue is with this method the input box does not appear at all, even on flushing cache. :thinking: